src/app/state/blogpost/blogpost.selectors.ts

```diff
@@ -1,9 +1,11 @@
-import { createFeatureSelector, createSelector } from "@ngrx/store";
-import { BlogpostState } from "./blogpost.state";
+import { createFeatureSelector, createSelector } from '@ngrx/store';
+import { BlogpostState } from './blogpost.state';
 
 const getBlogpostFeatureState = createFeatureSelector<BlogpostState>('posts');
 
-export const getPosts = createSelector(
-    getBlogpostFeatureState,
-    state => state.posts
-);
\ No newline at end of file
+export const getPosts = createSelector(getBlogpostFeatureState, (state) =>
+  Object.values(state.posts)
+);
+
+export const getPostById = (id: number) =>

```

Comment:
Keep in mind that new selector is generated on every call - this _might_ cause slowdowns in the application when user very often - for example table with thousand rows where each row would generate new selector

also this approach is not very friendly for selector composition and can get out of control very fast
e.g. 
export const getPostTitle = (id) => createSelector(getPostById(id), post => post.title)
// ...
postTitle$ = this.store.select(getPostTitle(id)) -  in this example we are calling createSelector twice on every use.


usually there are 2 alternatives:
1) keep activePost in the state
or
2) retrieve post in the component
this.post$ = this.store.select(getPosts).pipe(map((allPosts) => allPosts[this.id]))